rust/operator-binary/src/main.rs

```diff
@@ -71,34 +74,66 @@ async fn main() -> anyhow::Result<()> {
-            )
-            .owns(
-                watch_namespace.get_api::<Service>(&client),
-                ListParams::default(),
-            )
-            .owns(
-                watch_namespace.get_api::<StatefulSet>(&client),
-                ListParams::default(),
-            )
-            .shutdown_on_signal()
-            .run(
-                superset_controller::reconcile_superset,
-                superset_controller::error_policy,
-                Context::new(superset_controller::Ctx {
-                    client: client.clone(),
-                    product_config,
-                }),
-            )
-            .map(|res| {
-                report_controller_reconciled(
-                    &client,
-                    "supersetclusters.superset.stackable.tech",
-                    &res,
+            );
+            let superset_store = superset_controller_builder.store();
+            let superset_controller = superset_controller_builder
+                .owns(
+                    watch_namespace.get_api::<Service>(&client),
+                    ListParams::default(),
                 )
-            });
+                .owns(
+                    watch_namespace.get_api::<StatefulSet>(&client),
+                    ListParams::default(),
+                )
+                .shutdown_on_signal()
+                .watches(
+                    watch_namespace.get_api::<AuthenticationClass>(&client),
+                    ListParams::default(),
+                    move |authentication_class| {
+                        superset_store
+                            .state()
+                            .into_iter()
+                            .filter(move |superset: &Arc<SupersetCluster>| {
+                                match &superset.spec.authentication_config {
+                                    Some(authentication_config) => {
+                                        match &authentication_config.methods[..] {
+                                            [authentication_method]
+                                                if &authentication_method.authentication_class
+                                                    == authentication_class
+                                                        .metadata
+                                                        .name
+                                                        .as_ref()
+                                                        .unwrap() =>
+                                            {
+                                                true
+                                            }
+                                            _ => false,
+                                        }
+                                    }
+                                    None => false,
+                                }
+                            })
+                            .map(|superset| ObjectRef::from_obj(&*superset))
```

Comment:
Does it make sense to break this out into a separate function?

Reply:
Yeah i guess. Extracted it